### PR TITLE
chore!: update protons and connection encryption interface

### PR DESCRIPTION
**Pull request by mpetrunic**
BREAKING CHANGE: secureInbound and secureOutbound now accept Duplex instead of any

### CODE REVIEW

**Review by achingbrain — APPROVED** [review]

**Comment by wemeetagain** [comment]
this is a breaking change, changing the PR title

**Review by wemeetagain — DISMISSED** [review]

**Comment by mpetrunic** [comment]
> this is a breaking change, changing the PR title

What is breaking change here? 😅

File: package.json [hunk 1]
```diff
@@ -78,22 +78,23 @@
...
-    "@libp2p/interface-connection-encrypter-compliance-tests": "^1.0.1",
+    "@libp2p/interface-connection-encrypter-compliance-tests": "^2.0.1",
     "@libp2p/peer-id-factory": "^1.0.8",
     "aegir": "^37.3.0",
     "benchmark": "^2.1.4",
     "iso-random-stream": "^2.0.2",
     "mkdirp": "^1.0.4",
-    "protons": "^4.0.0",
-    "sinon": "^14.0.0"
+    "protons": "^5.0.0",
```
Comment (achingbrain) [inline package.json:95]:
```suggestion
    "protons": "^5.1.0",
```

File: package.json [hunk 2]
```diff
@@ -78,22 +78,23 @@
...
     "it-length-prefixed": "^8.0.2",
     "it-pair": "^2.0.2",
+    "it-stream-types": "^1.0.4",
     "it-pb-stream": "^2.0.1",
     "it-pipe": "^2.0.3",
-    "it-stream-types": "^1.0.4",
-    "protons-runtime": "^2.0.1",
-    "uint8arraylist": "^2.0.0",
-    "uint8arrays": "^3.0.0"
+    "protons-runtime": "^3.0.1",
```
Comment (achingbrain) [inline package.json:84]:
```suggestion
    "protons-runtime": "^3.1.0",
```

File: package.json [hunk 3]
```diff
@@ -78,22 +78,23 @@
     "@stablelib/x25519": "^1.0.1",
     "it-length-prefixed": "^8.0.2",
     "it-pair": "^2.0.2",
+    "it-stream-types": "^1.0.4",
     "it-pb-stream": "^2.0.1",
```
Comment (achingbrain) [inline package.json:82]:
```suggestion
    "it-pb-stream": "^2.0.2",
```

**Review by achingbrain — CHANGES_REQUESTED** [review]
This needs one final regeneration of the protobuf files with protons at `5.1.0` but then it should be good to go and will unblock https://github.com/libp2p/js-libp2p/pull/1335

**Pull request merged**